modules/cudaarithm/test/test_element_operations.cpp

```diff
@@ -2857,5 +2953,58 @@ INSTANTIATE_TEST_CASE_P(CUDA_Arithm, PolarToCart, testing::Combine(
     testing::Values(AngleInDegrees(false), AngleInDegrees(true)),
     WHOLE_SUBMAT));
 
+PARAM_TEST_CASE(PolarToCartInterleaved, cv::cuda::DeviceInfo, cv::Size, MatType, AngleInDegrees, UseRoi)
+{
+    cv::cuda::DeviceInfo devInfo;
+    cv::Size size;
+    int type;
+    bool angleInDegrees;
+    bool useRoi;
+
+    virtual void SetUp()
+    {
+        devInfo = GET_PARAM(0);
+        size = GET_PARAM(1);
+        type = GET_PARAM(2);
+        angleInDegrees = GET_PARAM(3);
+        useRoi = GET_PARAM(4);
+
+        cv::cuda::setDevice(devInfo.deviceID());
+    }
+};
+
+CUDA_TEST_P(PolarToCartInterleaved, Accuracy)
+{
+    cv::Mat magnitudeAngleChannels[2];
+    cv::Mat& magnitude = magnitudeAngleChannels[0];
+    cv::Mat& angle = magnitudeAngleChannels[1];
+    magnitude = randomMat(size, type);
+    angle = randomMat(size, type);
+    cv::Mat magnitudeAngle;
+    cv::merge(magnitudeAngleChannels, 2, magnitudeAngle);
+    const double tol = (type == CV_32FC1 ? 1.6e-4 : 1e-4) * (angleInDegrees ? 1.0 : 19.47);
```

Comment:
Again I would suggest a test case per function, comparing to `cv::polarToCart` especially if you are going to use these tolerances.  In this funciton you could now be 2*tol away from the CPU result.